[CI] BanFailureLoggingTests testLogsAtDebugOnDisconnectionDuringBan failing

Suspicious that this failed after this same test failed for a different reason a couple of weeks ago: https://github.com/elastic/elasticsearch/pull/75171

**Build scan:**
https://gradle-enterprise.elastic.co/s/prdb3lqjdshak/tests/:server:test/org.elasticsearch.tasks.BanFailureLoggingTests/testLogsAtDebugOnDisconnectionDuringBan

**Reproduction line:**
`./gradlew ':server:test' --tests "org.elasticsearch.tasks.BanFailureLoggingTests.testLogsAtDebugOnDisconnectionDuringBan" -Dtests.seed=3A306031869227FC -Dtests.locale=sr-BA -Dtests.timezone=America/Argentina/Rio_Gallegos -Druntime.java=16`

**Applicable branches:**
7.14

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.tasks.BanFailureLoggingTests&tests.test=testLogsAtDebugOnDisconnectionDuringBan

**Failure excerpt:**
```
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=231, name=elasticsearch[TransportTasksActionTests][management][T#1], state=RUNNABLE, group=TGRP-BanFailureLoggingTests]

  at __randomizedtesting.SeedInfo.seed([3A306031869227FC:8D572F744CBF6026]:0)

  Caused by: java.lang.AssertionError: (No message provided)

    at __randomizedtesting.SeedInfo.seed([3A306031869227FC]:0)
    at org.junit.Assert.fail(Assert.java:86)
    at org.junit.Assert.assertTrue(Assert.java:41)
    at org.junit.Assert.assertTrue(Assert.java:52)
    at org.elasticsearch.tasks.BanFailureLoggingTests.lambda$runTest$5(BanFailureLoggingTests.java:127)
    at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1035)
    at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1008)
    at org.elasticsearch.tasks.BanFailureLoggingTests.lambda$runTest$6(BanFailureLoggingTests.java:127)
    at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:61)
    at org.elasticsearch.transport.InboundHandler$1.doRun(InboundHandler.java:212)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:732)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
    at java.lang.Thread.run(Thread.java:831)

```

--- a/server/src/test/java/org/elasticsearch/tasks/BanFailureLoggingTests.java
+++ b/server/src/test/java/org/elasticsearch/tasks/BanFailureLoggingTests.java
@@ -37,7 +37,9 @@ import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.ReentrantLock;
 import java.util.function.Function;
 
 import static org.hamcrest.Matchers.anyOf;
@@ -98,6 +100,9 @@ public class BanFailureLoggingTests extends TaskManagerTestCase {
 
         try {
 
+            // the child task might not run, but if it does we must wait for it to be cancelled before shutting everything down
+            final ReentrantLock childTaskLock = new ReentrantLock();
+
             final MockTransportService parentTransportService = MockTransportService.createNewService(
                 Settings.EMPTY,
                 Version.CURRENT,
@@ -124,7 +129,13 @@ public class BanFailureLoggingTests extends TaskManagerTestCase {
                 },
                 (request, channel, task) -> {
                     final CancellableTask cancellableTask = (CancellableTask) task;
-                    assertBusy(() -> assertTrue(cancellableTask.isCancelled()));
+                    if (childTaskLock.tryLock()) {
+                        try {
+                            assertBusy(() -> assertTrue("task " + task.getId() + " should be cancelled", cancellableTask.isCancelled()));
+                        } finally {
+                            childTaskLock.unlock();
+                        }
+                    }
                     channel.sendResponse(new TaskCancelledException("task cancelled"));
                 });
 
@@ -161,13 +172,15 @@ public class BanFailureLoggingTests extends TaskManagerTestCase {
             final PlainActionFuture<Void> cancellationFuture = new PlainActionFuture<>();
             parentTransportService.getTaskManager().cancelTaskAndDescendants(parentTask, "test", true, cancellationFuture);
             try {
-                cancellationFuture.actionGet(TimeValue.timeValueSeconds(5));
+                cancellationFuture.actionGet(TimeValue.timeValueSeconds(10));
             } catch (NodeDisconnectedException e) {
                 // acceptable; we mostly ignore the result of cancellation anyway
             }
 
             // assert busy since failure to remove a ban may be logged after cancellation completed
             assertBusy(appender::assertAllExpectationsMatched);
+
+            assertTrue("child tasks did not finish in time", childTaskLock.tryLock(15, TimeUnit.SECONDS));
         } finally {
             Collections.reverse(resources);
             IOUtils.close(resources);

